irnl_rdt_correction/rdt_handling.py

```diff
@@ -0,0 +1,235 @@
+
+        main_name = f'C{ORDER_NAME_MAP[order]}{SKEW_NAME_MAP[skew]}X'
+        extra = "F" if accel == "hllhc" and ip in [1, 5] else ""
+        self.type = f'M{main_name}{extra}'
+        self.name = f'{self.type}.{POSITION:d}{side}{ip:d}'
+        self.circuit = f'K{main_name}{POSITION:d}.{side}{ip:d}'
+        self.strength_component = f"K{order-1}{SKEW_NAME_MAP[skew]}L"  # MAD-X order notation
+        self.value = 0
+
+    def __repr__(self):
+        return f"IRCorrector object {str(self)}"
+
+    def __str__(self):
+        return f"{self.name} ({self.field_component}), {self.strength_component}: {self.value: 6E}"
+
+    def __lt__(self, other):
+        if self.order == other.order:
+            if self.skew == other.skew:
+                if self.ip == other.ip:
+                    return (self.side == SIDES[1]) < (other.side == SIDES[1])
+                return self.ip < other.ip
+            return self.skew < other.skew
+        return self.order < other.order
+
+    def __gt__(self, other):
+        if self.order == other.order:
+            if self.skew == other.skew:
+                if self.ip == other.ip:
+                    return (self.side == SIDES[1]) > (other.side == SIDES[1])
+                return self.ip > other.ip
+            return self.skew > other.skew
+        return self.order > other.order
+
+    def __hash__(self):
+        return hash(self.name)
+
+    def __eq__(self, other):
+        return self.name == other.name
+
+
+class RDT:
+    """ Representation of Resonance Driving Term attributes. """
+    def __init__(self, name: str):
+        self.name = name
+        self.jklm = tuple(int(i) for i in name[1:5])
+        self.j, self.k, self.l, self.m = self.jklm
+        self.skew = bool((self.l + self.m) % 2)
+        self.order = sum(self.jklm)
+        self.swap_beta_exp = name.endswith("*")  # swap beta-exponents
+
+    def __repr__(self):
+        return f"RDT object {str(self)}"
+
+    def __str__(self):
+        return f"{self.name} ({order2field_component(self.order, self.skew)})"
+
+    def __lt__(self, other):
+        if self.order == other.order:
+            if self.skew == other.skew:
+                return len(self.name) < len(other.name)
+            return self.skew < other.skew
+        return self.order < other.order
+
+    def __gt__(self, other):
+        if self.order == other.order:
+            if self.skew == other.skew:
+                return len(self.name) > len(other.name)
+            return self.skew > other.skew
+        return self.order > other.order
+
+    def __hash__(self):
+        return hash(self.name)
+
+    def __eq__(self, other):
+        return self.name == other.name
+
+
+RDTMap = Dict[RDT, Sequence[str]]
+
+
+# RDT Sorting ------------------------------------------------------------------
+
+def sort_rdts(rdts: Sequence, rdts2: Sequence) -> Tuple[RDTMap, RDTMap]:
+    """ Sorts RDTs by reversed-order and orientation (skew, normal). """
```

Comment:
Arguments and returned in docstring. Could be nice to give an option for the order of the sorting or at least explain why it does reverse-order